src/app/space/create/deployments/deployments-donut/deployments-donut-chart/deployments-donut-chart.component.ts

```diff
@@ -0,0 +1,245 @@
+import { Component, DoCheck, Input, OnChanges, OnDestroy, OnInit, SimpleChanges } from '@angular/core';
+
+import { debounce, get, isEmpty, isEqual, reject, size, uniqueId } from 'lodash';
+
+import * as c3 from 'c3';
+import * as d3 from 'd3';
+
+@Component({
+  selector: 'deployments-donut-chart',
+  templateUrl: './deployments-donut-chart.component.html',
+  styleUrls: ['./deployments-donut-chart.component.less']
+})
+export class DeploymentsDonutChartComponent implements DoCheck, OnInit, OnChanges, OnDestroy {
+
+  @Input() pods: any;
+  @Input() mini: boolean;
+  @Input() desiredReplicas: number;
+  @Input() idled: boolean;
+
+  chartId = uniqueId('deployments-donut-chart');
+  podStatusData: any;
+  total: number;
+
+  private phases = ['Running',
+    'Not Ready',
+    'Warning',
+    'Error',
+    'Pulling',
+    'Pending',
+    'Succeeded',
+    'Terminating',
+    'Unknown'];
+
+  private config: any;
+  private chart: any;
+
+
+  private debounceUpdate: (countByPhase: any) => void;
+  private prevPodPhaseCount: any;
+
+  constructor() { }
+
+  ngOnInit(): void {
+    this.config = {
+      type: 'donut',
+      bindto: '#' + this.chartId,
+      donut: {
+        expand: false,
+        label: {
+          show: false
+        },
+        width: this.mini ? 5 : 10
+      },
+      size: {
+        height: this.mini ? 45 : 150,
+        width: this.mini ? 45 : 150
+      },
+      legend: {
+        show: false
+      },
+      tooltip: {
+        format: {
+          value: function (value, ratio, id) {
+            if (!value) {
+              return undefined;
+            }
+            if (id === 'Empty') {
+              return undefined;
+            }
+
+            return value;
+          }
+        }
+      },
+      transition: {
+        duration: 350
+      },
+      data: {
+        type: 'donut',
+        groups: [this.phases],
+        order: null,
+        colors: {
+          Empty: '#ffffff',
+          Running: '#00b9e4',
+          'Not Ready': '#beedf9',
+          Warning: '#f39d3c',
+          Error: '#d9534f',
+          Pulling: '#d1d1d1',
+          Pending: '#ededed',
+          Succeeded: '#3f9c35',
+          Terminating: '#00659c',
+          Unknown: '#f9d67a'
+        },
```

Comment:
Maybe some comments denoting what the colours are here, just roughly? ex "Dark blue" is specific enough IMO.